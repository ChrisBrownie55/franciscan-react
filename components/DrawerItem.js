import React, { Component } from 'react'
import { withStyles } from 'material-ui/styles'
import List, { ListItem, ListItemText } from 'material-ui/List'
import Collapse from 'material-ui/transitions/Collapse'
import ExpandLess from 'material-ui-icons/ExpandLess'
import ExpandMore from 'material-ui-icons/ExpandMore'
import Link from 'next/link'
import Grid from 'material-ui/Grid'

const styles = theme => ({
  nested: {
    paddingLeft: theme.spacing.unit * 4
  },
  nestedMore: {
    paddingLeft: theme.spacing.unit * 6
  },
  subLink: {
    textDecoration: 'none'
  },
  linkText: {
    color: 'rgba(0,0,0,0.54)'
  },
  subLinkText: {
    color: 'rgba(0,0,0,0.54)',
    fontWeight: 300
  },
  subItemIconContainer: {
    display: 'flex',
    justifyContent: 'center',
    borderLeft: '1px solid rgba(0,0,0,0.08)',
    cursor: 'pointer',
    '&:hover': {
      backgroundColor: 'rgba(0, 0, 0, 0.08)',
      borderLeft: 'none'
    }
  }
})

class DrawerItem extends Component {
  state = {
    drawerItems: {}
  }

  expandSubItem = itemId => {
    this.setState(prevState => ({
      drawerItems: {
        ...prevState.drawerItems,
        [itemId]: !prevState.drawerItems[itemId]
      }
    }))
    console.log(this.state.drawerItems)
  }
  render () {
    const { drawerItems } = this.state
    const {
      classes,
      toggleDrawer,
      primaryText,
      submenuItems,
      isOpen,
      expandItem,
      itemId
    } = this.props
    return (
      <div>
        <ListItem button onClick={() => expandItem(itemId)}>
          <ListItemText
            style={{ paddingLeft: '16px' }}
            inset
            primary={primaryText}
          />
          {isOpen ? <ExpandLess /> : <ExpandMore />}
        </ListItem>
        {submenuItems && (
          <Collapse component="li" in={isOpen} timeout="auto" unmountOnExit>
            <List disablePadding>
              {submenuItems.map(
                item =>
                  item.subMenu ? (
                    <Grid container spacing={0} key={item.text}>
                      <Grid
                        onClick={toggleDrawer}
                        onKeyDown={toggleDrawer}
                        item
                        xs={10}
                      >
                        <Link prefetch href={item.asUrl} as={item.linkUrl}>
                          <a className={classes.subLink}>
                            <ListItem button className={classes.nested}>
                              <ListItemText
                                inset
                                style={{ paddingLeft: '16px' }}
                                primary={item.text}
                                classes={{ primary: classes.linkText }}
                              />
                            </ListItem>
                          </a>
                        </Link>
                      </Grid>
                      <Grid
                        item
                        xs={2}
                        onClick={() => this.expandSubItem(item.text)}
                        className={classes.subItemIconContainer}
                      >
                        {drawerItems[item.text] ? (
                          <ExpandLess
                            style={{
                              alignSelf: 'center',
                              color: 'rgba(0,0,0,0.54'
                            }}
                          />
                        ) : (
                          <ExpandMore
                            style={{
                              alignSelf: 'center',
                              color: 'rgba(0,0,0,0.54'
                            }}
                          />
                        )}
                      </Grid>
                      <Collapse
                        component="li"
                        in={drawerItems[item.text]}
                        timeout="auto"
                        unmountOnExit
                      >
                        <List disablePadding>
                          {item.subMenu.map(subItem => (
                            <div
                              onClick={toggleDrawer}
                              onKeyDown={toggleDrawer}
                              key={subItem.text}
                            >
                              <Link
                                prefetch
                                href={subItem.asUrl}
                                as={subItem.linkUrl}
                              >
                                <a className={classes.subLink}>
                                  <ListItem
                                    button
                                    className={classes.nestedMore}
                                  >
                                    <ListItemText
                                      inset
                                      style={{ paddingLeft: '16px' }}
                                      primary={subItem.text}
                                      classes={{ primary: classes.subLinkText }}
                                    />
                                  </ListItem>
                                </a>
                              </Link>
                            </div>
                          ))}
                        </List>
                      </Collapse>
                    </Grid>
                  ) : (
                    <div
                      onClick={toggleDrawer}
                      onKeyDown={toggleDrawer}
                      key={item.text}
                    >
                      <Link prefetch href={item.asUrl} as={item.linkUrl}>
                        <a className={classes.subLink}>
                          <ListItem button className={classes.nested}>
                            <ListItemText
                              inset
                              style={{ paddingLeft: '16px' }}
                              primary={item.text}
                              classes={{ primary: classes.linkText }}
                            />
                          </ListItem>
                        </a>
                      </Link>
                    </div>
                  )
              )}
            </List>
          </Collapse>
        )}
      </div>
    )
  }
}

export default withStyles(styles)(DrawerItem)
