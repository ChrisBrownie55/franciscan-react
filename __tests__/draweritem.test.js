/* eslint-env jest */
import React from 'react'
// import sinon from 'sinon'
import { mount } from 'enzyme'

import DrawerItem from '../components/DrawerItem'

describe('DrawerItem', () => {
  it('should be defined', () => {
    expect(DrawerItem).toBeDefined()
  })
  it('should render', () => {
    const wrapper = mount(<DrawerItem />)
    expect(wrapper).toMatchSnapshot()
  })
  // it('simulates click events', () => {
  //   const onButtonClick = sinon.spy()
  //   const wrapper = mount(<DrawerItem expandItem={onButtonClick} />)
  //   wrapper.find('button').simulate('click')
  //   expect(wrapper.state().isOpen).toEqual(true)
  // })
})
