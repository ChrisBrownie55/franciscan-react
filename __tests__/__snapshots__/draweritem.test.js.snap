// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerItem should render 1`] = `
<WithStyles(DrawerItem)>
  <DrawerItem
    classes={
      Object {
        "linkText": "DrawerItem-linkText-4",
        "nested": "DrawerItem-nested-1",
        "nestedMore": "DrawerItem-nestedMore-2",
        "subItemIconContainer": "DrawerItem-subItemIconContainer-6",
        "subLink": "DrawerItem-subLink-3",
        "subLinkText": "DrawerItem-subLinkText-5",
      }
    }
  >
    <div>
      <WithStyles(ListItem)
        button={true}
        onClick={[Function]}
      >
        <ListItem
          ContainerComponent="li"
          button={true}
          classes={
            Object {
              "button": "MuiListItem-button-15",
              "container": "MuiListItem-container-8",
              "default": "MuiListItem-default-10",
              "dense": "MuiListItem-dense-11",
              "disabled": "MuiListItem-disabled-12",
              "divider": "MuiListItem-divider-13",
              "gutters": "MuiListItem-gutters-14",
              "keyboardFocused": "MuiListItem-keyboardFocused-9",
              "root": "MuiListItem-root-7",
              "secondaryAction": "MuiListItem-secondaryAction-16",
            }
          }
          dense={false}
          disableGutters={false}
          disabled={false}
          divider={false}
          onClick={[Function]}
        >
          <WithStyles(ButtonBase)
            className="MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15"
            classes={
              Object {
                "keyboardFocused": "MuiListItem-keyboardFocused-9",
              }
            }
            component="div"
            disabled={false}
            onClick={[Function]}
          >
            <ButtonBase
              centerRipple={false}
              className="MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15"
              classes={
                Object {
                  "disabled": "MuiButtonBase-disabled-18",
                  "keyboardFocused": "MuiButtonBase-keyboardFocused-19 MuiListItem-keyboardFocused-9",
                  "root": "MuiButtonBase-root-17",
                }
              }
              component="div"
              disableRipple={false}
              disabled={false}
              focusRipple={false}
              onClick={[Function]}
              tabIndex="0"
              type="button"
            >
              <div
                className="MuiButtonBase-root-17 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <WithStyles(ListItemText)
                  inset={true}
                  key=".0"
                  style={
                    Object {
                      "paddingLeft": "16px",
                    }
                  }
                >
                  <ListItemText
                    classes={
                      Object {
                        "dense": "MuiListItemText-dense-22",
                        "inset": "MuiListItemText-inset-21",
                        "primary": "MuiListItemText-primary-23",
                        "root": "MuiListItemText-root-20",
                        "secondary": "MuiListItemText-secondary-24",
                        "textDense": "MuiListItemText-textDense-25",
                      }
                    }
                    disableTypography={false}
                    inset={true}
                    style={
                      Object {
                        "paddingLeft": "16px",
                      }
                    }
                  >
                    <div
                      className="MuiListItemText-root-20 MuiListItemText-inset-21"
                      style={
                        Object {
                          "paddingLeft": "16px",
                        }
                      }
                    />
                  </ListItemText>
                </WithStyles(ListItemText)>
                <pure(ExpandMore)
                  key=".1"
                >
                  <ExpandMore>
                    <WithStyles(SvgIcon)>
                      <SvgIcon
                        classes={
                          Object {
                            "colorAction": "MuiSvgIcon-colorAction-29",
                            "colorDisabled": "MuiSvgIcon-colorDisabled-30",
                            "colorError": "MuiSvgIcon-colorError-31",
                            "colorPrimary": "MuiSvgIcon-colorPrimary-27",
                            "colorSecondary": "MuiSvgIcon-colorSecondary-28",
                            "root": "MuiSvgIcon-root-26",
                          }
                        }
                        color="inherit"
                        viewBox="0 0 24 24"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root-26"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          />
                        </svg>
                      </SvgIcon>
                    </WithStyles(SvgIcon)>
                  </ExpandMore>
                </pure(ExpandMore)>
                <WithStyles(TouchRipple)
                  center={false}
                  innerRef={[Function]}
                >
                  <TouchRipple
                    center={false}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child-36",
                        "childLeaving": "MuiTouchRipple-childLeaving-37",
                        "childPulsate": "MuiTouchRipple-childPulsate-38",
                        "ripple": "MuiTouchRipple-ripple-33",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate-35",
                        "rippleVisible": "MuiTouchRipple-rippleVisible-34",
                        "root": "MuiTouchRipple-root-32",
                      }
                    }
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      className="MuiTouchRipple-root-32"
                      component="span"
                      enter={true}
                      exit={true}
                    >
                      <span
                        className="MuiTouchRipple-root-32"
                      />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </div>
            </ButtonBase>
          </WithStyles(ButtonBase)>
        </ListItem>
      </WithStyles(ListItem)>
    </div>
  </DrawerItem>
</WithStyles(DrawerItem)>
`;
